consensus/istanbul/backend/backend.go

```diff
@@ -313,7 +338,35 @@ func (sb *Backend) Verify(proposal istanbul.Proposal) (time.Duration, error) {
 		return 0, errInvalidCoinbase
 	}
 
-	// TODO: Check parent signatures
+	number := block.Number().Uint64()
+	// Only check the parent seals for blocks after the first block, since its
+	// only parent is the genesis block and is not expected to have a valid bls
+	// signature
+	if number > 1 {
+		// we need the snapshot from the PREVIOUS block, so we need to get blockNumber-2
+		snap, err := sb.snapshot(sb.chain, block.Number().Uint64()-2, block.ParentHash(), nil)
+		if err != nil {
+			return 0, err
+		}
+
+		// Get the validator set from the previous block
+		validators := snap.ValSet.Copy()
+
+		// Get the parent seals for this block (ie signature & bitmap)
+		parentSeal := block.ParentSeal()
+
+		// Get the parent block's proposal seal
+		proposalSeal := istanbulCore.PrepareCommittedSeal(block.ParentHash())
+
+		fmt.Println("Proposal Seal:", common.ToHex(proposalSeal))
```

Comment:
s/fmt.Println/sb.logger.X everywhere for any log messages you want to keep around

Reply:
PR still needs some extra work done, will be cleaning it up from all debugging related logs/println's I have added.